lib/twitter/rest/account_activity.rb

```diff
@@ -118,17 +118,65 @@ def check_subscription(env_name_or_webhook_id)
       # @rate_limited Yes
-      # @authentication Requires user context - all consumer and access tokens
+      # @authentication Workes with user context and application context
       # @raise [Twitter::Error::Unauthorized] Error raised when supplied user credentials are not valid.
       # @return [nil]
       # @param env_name_or_webhook_id [String] Environment Name or Webhook Id
-      def deactivate_subscription(env_name_or_webhook_id)
+      # @param user_id [String] Id of the user to unsubscribe
+      def deactivate_subscription(env_name_or_webhook_id, user_id: nil)
+        user_token? ? deactivate_with_user_token(env_name_or_webhook_id) : deactivate_with_app_token(env_name_or_webhook_id, user_id)
+      end
+
+      # Returns the count of subscriptions that are currently active on your account for all activities.
+      # @see https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/api-reference/aaa-premium#get-account-activity-all-subscriptions-count
+      # @see https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/api-reference/aaa-enterprise#get-account-activity-all-subscriptions-count
+      # @note Gives the number of subscriptions to an app
+      # @rate_limited yes
+      # @authentication Requires application context
+      # @raise [Twitter::Error::Unauthorized] Error raised when supplied user credentials are not valid.
+      # @return [Hash]
+      def count_subscriptions
+        if enterprise_api
+          perform_request(:get, '/1.1/account_activity/subscriptions/count.json')
+        else
+          perform_request(:get, '/1.1/account_activity/all/subscriptions/count.json')
+        end
+      end
+
+      # Returns a list of the current All Activity type subscriptions.
+      # @see https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/api-reference/aaa-premium#get-account-activity-all-env-name-subscriptions-list
+      # @see https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/api-reference/aaa-enterprise#get-account-activity-all-env-name-subscriptions-list
+      # @note Gives list of subscriptions to an app
+      # @rate_limited yes
+      # @authentication Requires application context
+      # @raise [Twitter::Error::Unauthorized] Error raised when supplied user credentials are not valid.
+      # @return [Hash]
+      # @param env_name_or_webhook_id [String] Environment Name or Webhook Id
+      def list_subscriptions(env_name_or_webhook_id)
+        if enterprise_api
+          perform_request(:get, "/1.1/account_activity/webhooks/#{env_name_or_webhook_id}/subscriptions/all/list.json")
+        else
+          perform_request(:get, "/1.1/account_activity/all/#{env_name_or_webhook_id}/subscriptions/list.json")
+        end
+      end
+
+    private
+
+      def deactivate_with_user_token(env_name_or_webhook_id)
```

Comment:
These endpoints are [deprecated](https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/api-reference). We shouldn't use them.

It was even [supposed to be retired](https://developer.twitter.com/en/docs/changelog/standard-premium) on January 15, 2020.

Reply:
I propose to add a deprecation warning instead of removing it because it continues to work.

Reply:
I removed deprecated endpoints.